test/gtest/core/test_alltoall.cc

```diff
@@ -64,6 +64,66 @@ class test_alltoall : public UccCollArgs, public ucc::test
             }
         }
     }
+    void data_init(int nprocs, UccJob *job, ucc_datatype_t dtype,
+                   size_t single_rank_count, UccCollCtxVec &ctxs)
+    {
+        void *sbuf;
+        void *rbuf;
+        long *work_buf;
+
+        ctxs.resize(nprocs);
+        for (auto i = 0; i < nprocs; i++) {
+            ucc_coll_args_t *coll =
+                (ucc_coll_args_t *)calloc(1, sizeof(ucc_coll_args_t));
+
+            ctxs[i] =
+                (gtest_ucc_coll_ctx_t *)calloc(1, sizeof(gtest_ucc_coll_ctx_t));
+            ctxs[i]->args = coll;
+
+            sbuf        = job->procs[i]->onesided_buf[0];
```

Comment:
Is this correct - taking procs[i] ?
The team being tested can be arbitrary subset of Job: e.g., job size is 15 ranks, and team consists of ranks (1,3,5,7,9). Then, the nprocs here is team size = 5, but actual processes from job should be job->procs[1], [3], [5], [7], [9] and so on.

Reply:
You're correct, the code was incorrect. 

I've updated the code to use the team instead of the job. I believe that should be correct.